src/metatrain/utils/data/readers/readers.py

```diff
@@ -135,28 +75,51 @@ def read_systems(
         determined from the file extension.
     :returns: list of systems stored in double precision
     """
-    return _base_reader(target="systems", filename=filename, reader=reader)
+    return _call_base_systems_reader(filename=filename, reader=reader)
 
 
-def read_virial(
-    filename: str,
-    target_value: str = "virial",
-    reader: Optional[str] = None,
-) -> List[TensorBlock]:
-    """Read virial informations from a file.
+def _call_base_target_reader(
+    target: DictConfig, target_kind: str
+) -> Tuple[List[TensorMap], TargetInfo]:
+    reader = target["reader"]
+    filename = target["read_from"]
 
-    :param filename: name of the file to read
-    :param target_value: target value key name to be parsed from the file.
-    :param reader: reader library for parsing the file. If :py:obj:`None` the library is
-        is tried to determined from the file extension.
-    :returns: virial stored in double precision as a :class:`metatensor.TensorBlock`
-    """
-    return _base_reader(
-        target="virial",
-        filename=filename,
-        reader=reader,
-        key=target_value,
-    )
+    if reader is None:
+        try:
+            file_suffix = Path(filename).suffix
+            reader = DEFAULT_READER[file_suffix]
+        except KeyError:
+            raise ValueError(
+                f"File extension {file_suffix!r} is not linked to a default reader "
+                "library. You can try reading it by setting a specific 'reader' from "
+                f"the known ones: {', '.join(AVAILABLE_READERS)} "
+            )
+
+    try:
+        reader_mod = importlib.import_module(
+            name=f".{reader}", package="metatrain.utils.data.readers"
+        )
+    except ImportError:
+        raise ValueError(
+            f"Reader library {reader!r} not supported. Choose from "
+            f"{', '.join(AVAILABLE_READERS)}"
+        )
+
+    try:
+        reader_met = getattr(reader_mod, f"read_{target_kind}")
+    except AttributeError:
+        raise ValueError(f"Reader library {reader!r} can't read {target!r}.")
```

Comment:
This seems not tested and should we give a suggestion to the user what they can do?

Reply:
The suggestion is there. The test can't be done easily because both ASE and metatensor can read all types of targets

Reply:
If they can read all errors why is there an error?